docs/build/guides/conventions/working-with-contract-specs.mdx

```diff
@@ -0,0 +1,1264 @@
+    )  {}
+}
+```
+
+```json
+[
+  {
+    "type": "union",
+    "doc": "",
+    "name": "TimeBoundKind",
+    "cases": [
+      {
+        "doc": "",
+        "name": "Before",
+        "values": []
+      },
+      {
+        "doc": "",
+        "name": "After",
+        "values": []
+      }
+    ]
+  },
+  {
+    "type": "struct",
+    "doc": "",
+    "name": "TimeBound",
+    "fields": [
+      {
+        "doc": "",
+        "name": "kind",
+        "value": {
+          "type": "custom",
+          "name": "TimeBoundKind"
+        }
+      },
+      {
+        "doc": "",
+        "name": "timestamp",
+        "value": {
+          "type": "u64"
+        }
+      }
+    ]
+  },
+  {
+    "type": "function",
+    "doc": "",
+    "name": "deposit",
+    "inputs": [
+      {
+        "doc": "",
+        "name": "from",
+        "value": {
+          "type": "address"
+        }
+      },
+      {
+        "doc": "",
+        "name": "token",
+        "value": {
+          "type": "address"
+        }
+      },
+      {
+        "doc": "",
+        "name": "amount",
+        "value": {
+          "type": "i128"
+        }
+      },
+      {
+        "doc": "",
+        "name": "claimants",
+        "value": {
+          "type": "vec",
+          "element": {
+            "type": "address"
+          }
+        }
+      },
+      {
+        "doc": "",
+        "name": "time_bound",
+        "value": {
+          "type": "custom",
+          "name": "TimeBound"
+        }
+      }
+    ],
+    "outputs": []
+  }
+]
+```
+
+```python
+from stellar_sdk import scval
+secret = "SXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+rpc_server_url = "https://soroban-testnet.stellar.org:443"
+contract_id = "CAIKIZOT2LXM2WBEPGTZTPHHTGVHGLEOSI4WE6BOHWIBHJOKHPMCOPLO"
+
+network_passphrase = Network.TESTNET_NETWORK_PASSPHRASE
+
+kp = Keypair.from_secret(secret)
+soroban_server = SorobanServer(rpc_server_url)
+source = soroban_server.load_account(kp.public_key)
+
+# Let's build a transaction that invokes the `deposit` function.
+tx = (
+    TransactionBuilder(source, network_passphrase, base_fee=1000)
+    .set_timeout(300)
+    .append_invoke_contract_function_op(
+        contract_id=contract_id,
+        function_name="deposit",
+        parameters=[
+            scval.to_address(kp.public_key),
+            scval.to_address("GXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"),
+            scval.to_int128(1),
+            scval.to_vec(
+                [
+                    scval.to_address("GXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"),
+                ]
+            ),
+            scval.to_map(
+                {
+                    scval.to_symbol("kind"): scval.to_vec(
+                        [
+                            scval.to_symbol("Before"),
+                        ]
+                    ),
+                    scval.to_symbol("timestamp"): scval.to_uint64(12346),
+                }
+            ),
+        ],
+    )
+    .build()
+)
+# tx.sign(kp)
```

Comment:
Is this meant to be here?

Reply:
I used it to test my implementation when writing the guide.
Taking it away now